Add crawler for Cayman Islands Legislative Assembly

aivuk opened the pull request:
Related issue: #364

The following crawler gets the elected members from the Cayman Island Legislative Assembly. The first attempt was to parse the html using xmltree, but the html is malformed as you can see in the snippet below on the `<h3><li>` mixup instead of `<li><h3>`:


```
                        <li>
                            <h3>Hon. Andre M. Ebanks, MP,</h3> Minister for Financial Services &amp; Commerce and Minister for Investment, Innovation &amp; Social Development and Member of Parliament for West Bay South
                        </li>
                        <h3>
                            <li>Hon. Sabrina Turner, MP,
                        </li></h3> Minister for Health &amp; Wellness and Member of Parliament for Prospect
                        
                        <li>
                            <h3>Hon. Johany S. "Jay" Ebanks, MP,</h3> Minister for Planning, Agriculture, Housing, &amp; Infrastructure and Member of Parliament for North Side
                        </li>
```
 
The code convert the html page to text using the method .text_content(), parse its content and deals with details as:

 * there is almost a pattern in the positions of a member. Sometimes the different positions are separated by 'and', sometimes by space as "Premier and Minister for Sustainability and Climate Resiliency and Finance & Economic Development Member of Parliament for Newlands''.
 * most of the Minister information with multiple areas have a & as separator "Minister for Tourism & Ports", but there are exceptions as "Minister for Sustainability and Climate Resiliency"
 *  removes some additional information from the member's names as 'MP', 'Hon. ', etc

jbothma commented on datasets/ky/legislativeassembly/crawler.py:
```diff
@@ -0,0 +1,128 @@
+    for person in persons:
+        person_proxy = context.make("Person")
+        h.apply_name(person_proxy, full=person["name"])
+        # TODO: Make a better ID
+        person_proxy.id = "legislativeassembly-{}".format(
+            person["name"].replace(" ", "_")
+        )
```
You could use https://zavod.opensanctions.org/context/#zavod.context.Context.make_id which should automatically generate `ky-la`-prefixed valid IDs for you.

These warnings are due to capitals in the current IDs 
```
2023-09-04 12:38:09 [warning  ] Rejected property value        [zavod.entity] dataset=ky_legislativeassembly entity_id=ky-la-fa6e409ac684162830bbc491b591a1666feeaef3 prop=holder value=legislativeassembly-Wayne_Panton
```

jbothma commented on datasets/ky/legislativeassembly/ky_legislativeassembly.yml:
```diff
@@ -0,0 +1,25 @@
+      Town North, George Town South, George Town West, Prospect, Red Bay, Bodden
+      Town East, Bodden Town West, Newlands, Savannah, North Side, East End, Cayman
+      Brac East, Cayman Brac West & Little Cayman.
+    acronym: ky-legislative
```
You can drop acronym - I think the acronymn would more accurately be CILA but I can't see that being a common term actually used by anyone and adding value

jbothma commented:
very nice!

What might be nice, if it's not a lot of work, is if the Member of parliament... position was only named `Member of parliament` without the constituency, just to limit the explosion of position names we'll be generating. The constituency name isn't important here.

Also add the source page as sourceUrl then users can easily go and see the original page if they like.

Finally, I'm not really sure how this will fail, but I wonder if there's a check you can add so that if parsing fails partially, we'd get alerted by warning logs. Complete failure would become a warning because it would generate no entities. But if some entities are generated, and then some weird text gets interpreted as positions, that's something we should try to actively guard against and rather er on the side of emitting context.log.warning(s) and outputting less data, than outputting broken data.

aivuk commented:
> What might be nice, if it's not a lot of work, is if the Member of parliament... position was only named `Member of parliament` without the constituency, just to limit the explosion of position names we'll be generating. The constituency name isn't important here.

I did change as you requested.
 
> Also add the source page as sourceUrl then users can easily go and see the original page if they like.

I added this to the persons entities. Do you want me to add to all entities (position and occupancies)?

> Finally, I'm not really sure how this will fail, but I wonder if there's a check you can add so that if parsing fails partially, we'd get alerted by warning logs. Complete failure would become a warning because it would generate no entities. But if some entities are generated, and then some weird text gets interpreted as positions, that's something we should try to actively guard against and rather er on the side of emitting context.log.warning(s) and outputting less data, than outputting broken data.

I added some sanity checks, if the initial marker that I'm using in the page is not found (the text "You are here") or if some of the positions blocks ("Premier", "Deputy Premier", etc). For checking if a weird text was interpreted as position I can only think in doing some pattern check, like testing if the position text is some of the list `positions` or if it's starts with "Member " or "Minister for ". I don't know if it's worthwhile  to create regex to test for valid "Minister for", what do you think?

jbothma commented:
This looks great.

Last think I spotted before merging, in the page I count 21 people but only 9 targets in the data.

jbothma commented on datasets/ky/legislativeassembly/ky_legislativeassembly.yml:
```diff
@@ -0,0 +1,25 @@
+name: ky_legislativeassembly
+title: "legislativeassembly"
```
Cayman Islands Legislative Assembly

jbothma replied:
woops, this was meant to be a "suggestion"

aivuk replied:
done

aivuk commented:
> This looks great.
> 
> Last think I spotted before merging, in the page I count 21 people but only 9 targets in the data.

Because you asked to not get them all here https://github.com/opensanctions/opensanctions/issues/364#issuecomment-1697244802 Or maybe I misunderstood you?

> no need to capture back bench, ex officio, oppostion, independent, etc

aivuk commented:
> Oh sorry, I meant we don't need to be able to distinguish whether someone is "backbench MP" from "ex oficio MP" - we're happy just calling them all "Member of parliament" if that's the title they have beside their name.
> 
> I totally see why you understood what you did from what I said.

ok, more work that I was expecting, but now in my last commit I'm getting all 21 persons from the legislative page. I also simplified the positions from "Member of Parliament for ..." to just "Member of Parliament".
